general/pionerv1_price.js

```diff
@@ -0,0 +1,177 @@
+
+                };
+        }
+    },
+
+    signParams: function (request, result) {
+        const signTime = result.oldestTimestamp ? result.oldestTimestamp : Math.floor(Date.now() / 1000);
+
+        switch (request.method) {
+            case 'price':
+                return [
+                    { name:'asset1' ,type: 'bytes32', value: result.asset1 },
+                    { name:'asset2' ,type: 'bytes32', value: result.asset2 },
+                    { name:'pairBid' ,type: 'uint256', value: result.pairBid },
+                    { name:'pairAsk' ,type: 'uint256', value: result.pairAsk },
+                    { name:'confidence' ,type: 'uint256', value: result.confidence },
+                    { name:'signTime' ,type: 'uint256', value: signTime.toString() }
+                ];
+        }
+    },
+
+fetchPrices: async function (asset1, asset2) {
+    const [result1, result2] = await Promise.all([
+        this.fetchAssetPrices(asset1),
+        this.fetchAssetPrices(asset2)
+    ]);
+
+    const adjustedPrices1 = asset1 === 'hardusd' ? { avgBid: 1, avgAsk: 1 } : this.calculateAveragePrices(result1.prices);
+    const adjustedPrices2 = asset2 === 'hardusd' ? { avgBid: 1, avgAsk: 1 } : this.calculateAveragePrices(result2.prices);
+    const asset1Confidence = asset1 === 'hardusd' ? 0 : this.calculateConfidence(result1.prices);
+    const asset2Confidence = asset2 === 'hardusd' ? 0 : this.calculateConfidence(result2.prices);
+    const highestConfidence = Math.max(asset1Confidence, asset2Confidence);
+    const oldestTimestamp = Math.min(result1.oldestTimestamp, result2.oldestTimestamp);
+
+    return {
+        pairBid: adjustedPrices1.avgBid / adjustedPrices2.avgBid,
+        pairAsk: adjustedPrices1.avgAsk / adjustedPrices2.avgAsk,
+        confidence: Math.max(1 - highestConfidence / 100, 0),
+        oldestTimestamp
+    };
+},
+
+    fetchAssetPrices: async function (asset) {
+        if (asset === 'hardusd') return { prices: [{ bid: 1, ask: 1, timestamp: null }], oldestTimestamp: null };
+        const [assetType, assetSymbol] = asset.split('.');
+        const apiConfigs = this.loadApiConfigsForType(assetType);
+        let oldestTimestamp = Infinity;
+        const prices = [];
+
+        for (const config of apiConfigs) {
+            const formattedSymbol = this.formatSymbolForAPI(assetType, assetSymbol);
+            const priceData = await this.fetchPriceFromAPI(config, formattedSymbol);
+            if (priceData) {
+                prices.push(priceData);
+                if (priceData.timestamp && priceData.timestamp < oldestTimestamp) {
+                    oldestTimestamp = priceData.timestamp;
+                }
+            }
+        }
+        return { prices, oldestTimestamp };
+    },
+
+    loadApiConfigsForType: function (assetType) {
+        const configs = [];
+        const prefix = `API_${assetType.toUpperCase()}_`;
+        for (const [key, value] of Object.entries(process.env)) {
```

Comment:
process.env is not yet isolated for apps. Looping over all env variables is insecure.
You can use something like this:

APPS_PIONERV1_VARS= _a_json_str_

and then decode the JSON string in the code